docs/docs-content/deployment-modes/agent-mode/overlay-preparation.md

```diff
@@ -0,0 +1,348 @@
+   ```
+
+2. Mask the `systemd-networkd-wait-online` service to prevent the system from hanging during boot if interfaces are not
+   immediately available.
+
+   ```yaml
+   #cloud-config
+   stages:
+     initramfs:
+       - name: "Mask systemd-networkd-wait-online"
+         commands:
+           - systemctl mask systemd-networkd-wait-online.service
+   ```
+
+3. Add a `systemd-networkd` configuration for interfaces with names starting with `en*` to enable DHCP.
+
+   ```yaml {13}
+   #cloud-config
+   stages:
+     initramfs:
+       - name: "Create systemd-networkd config for en* interfaces"
+         if: "[ ! -f /etc/systemd/network/20-dhcp.network ]"
+         files:
+           - path: /etc/systemd/network/20-dhcp.network
+             permissions: 0644
+             owner: 0
+             group: 0
+             content: |
+               [Match]
+               Name=en*
+
+               [Network]
+               DHCP=yes
+
+               [DHCP]
+               ClientIdentifier=mac
+   ```
+
+4. Add a fallback configuration for interfaces with names starting with `eth*`, to support legacy systems or naming
+   schemes.
+
+   ```yaml
+   #cloud-config
+   stages:
+     initramfs:
+       - name: "Create systemd-networkd config for eth* interfaces"
+         if: "[ ! -f /etc/systemd/network/20-dhcp-legacy.network ]"
+         files:
+           - path: /etc/systemd/network/20-dhcp-legacy.network
+             permissions: 0644
+             owner: 0
+             group: 0
+             content: |
+               [Match]
+               Name=eth*
+
+               [Network]
+               DHCP=yes
+
+               [DHCP]
+               ClientIdentifier=mac
+   ```
+
+5. Enable and restart the `systemd-networkd` service so it takes over interface management.
+
+   ```yaml
+   #cloud-config
+   stages:
+     initramfs:
+       - name: "Enable and restart systemd-networkd"
+         commands:
+           - systemctl enable systemd-networkd.service
+           - systemctl restart systemd-networkd.service
+   ```
+
+6. Enable and restart `systemd-resolved` to handle DNS resolution consistently with `systemd`.
+
+   ```yaml
+   #cloud-config
+   stages:
+     initramfs:
+       - name: "Enable and restart systemd-resolved"
+         commands:
+           - systemctl enable systemd-resolved.service
+           - systemctl restart systemd-resolved.service
+   ```
+
+7. Ensure `/etc/resolv.conf` points to the DNS configuration managed by `systemd-resolved`.
+
+   ```yaml
+   #cloud-config
+   stages:
+     initramfs:
+       - name: "Link systemd-resolved DNS to /etc/resolv.conf"
+         if: '[ ! -L /etc/resolv.conf ] || [ "$(readlink -f /etc/resolv.conf)" != "/run/systemd/resolve/resolv.conf" ]'
+         commands:
+           - ln -sf /run/systemd/resolve/resolv.conf /etc/resolv.conf
+   ```
+
+</TabItem>
+
+</Tabs>
+
+## Validate
+
+Use the following steps to ensure that your DNS is now managed by `systemd-networkd`.
```

Comment:
Maybe we should say "Use the following steps to ensure that your Network is now managed by `systemd-networkd`."